colorDepth returns '0' on detached frame

annevk opened the pull request:
#### 34925ea7ad8031c79323a34e6b93b03a5e83e1d0
<pre>
colorDepth returns &apos;0&apos; on detached frame
<a href="https://bugs.webkit.org/show_bug.cgi?id=265021">https://bugs.webkit.org/show_bug.cgi?id=265021</a>

Reviewed by Simon Fraser.

The specification requires 24 as a fallback rather than 0:
<a href="https://drafts.csswg.org/cssom-view/#dom-screen-colordepth">https://drafts.csswg.org/cssom-view/#dom-screen-colordepth</a>

Always returning 24 is tempting, but maybe not correct long term.

This synchronizes web-platforms-tests for css/cssom-view with:
<a href="https://github.com/web-platform-tests/wpt/commit/46476776fdbf0a62522621002ba8af68e4d478b6">https://github.com/web-platform-tests/wpt/commit/46476776fdbf0a62522621002ba8af68e4d478b6</a>

* LayoutTests/http/tests/dom/same-origin-detached-window-properties-expected.txt:
* LayoutTests/http/tests/dom/same-origin-detached-window-properties.html:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/screen-detached-frame-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/screen-detached-frame.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/window-scrollBy-display-change-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/window-scrollBy-display-change-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/window-scrollBy-display-change.html: Added.
* Source/WebCore/page/Screen.cpp:
(WebCore::Screen::colorDepth const):

Canonical link: <a href="https://commits.webkit.org/270952@main">https://commits.webkit.org/270952@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/427401c7a164222c9566fa80dcbb589edccfab79

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26892 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5508 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/28128 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/29103 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24582 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/27338 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7357 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2899 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24465 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/27154 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4331 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/23086 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3795 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3853 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/24083 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29738 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24536 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24487 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/30080 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3873 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2073 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27990 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5331 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/23819 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6460 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4340 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4241 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/page/Screen.cpp:
```diff
@@ -92,7 +92,7 @@ unsigned Screen::colorDepth() const
     RefPtr frame = this->frame();
     if (!frame)
-        return 0;
+        return 24;
```
I wonder if it's worth codifying this default value as

```cpp
static constexpr uint32_t CSSOMDefaultColorDepth = 24;
```
in `Screen.h` and sharing between `Screen::colorDepth` and `PlatformScreenIOS::colorDepth`?

annevk replied:
The iOS value is really the platform value (though it's a constant for "reasons") and this is the universal fallback value so I think keeping them distinct makes sense.

However, if `colorDepth` is not an adequate API long term for HDR and such we might consider always returning 24 and not having the platform implementations of this API.

aprotyas reviewed (approved):
Non-reviewer LGTM

annevk committed:
colorDepth returns '0' on detached frame
https://bugs.webkit.org/show_bug.cgi?id=265021

Reviewed by Simon Fraser.

The specification requires 24 as a fallback rather than 0:
https://drafts.csswg.org/cssom-view/#dom-screen-colordepth

Always returning 24 is tempting, but maybe not correct long term.

This synchronizes web-platforms-tests for css/cssom-view with:
https://github.com/web-platform-tests/wpt/commit/46476776fdbf0a62522621002ba8af68e4d478b6

* LayoutTests/http/tests/dom/same-origin-detached-window-properties-expected.txt:
* LayoutTests/http/tests/dom/same-origin-detached-window-properties.html:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/screen-detached-frame-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/screen-detached-frame.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/window-scrollBy-display-change-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/window-scrollBy-display-change-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/window-scrollBy-display-change.html: Added.
* Source/WebCore/page/Screen.cpp:
(WebCore::Screen::colorDepth const):

Canonical link: https://commits.webkit.org/270952@main
